### PR TITLE
6198 include arMinDate and arMaxDate with the covid extension

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#6198

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/resources/business/business_tasks.py [hunk 1]
```diff
@@ -140,56 +135,44 @@ def construct_task_list(business):  # pylint: disable=too-many-locals; only 2 ex
...
-                    .get('annualGeneralMeetingDate', None)
-            if not last_ar_date:
-                last_ar_date = annual_report_filings[0].filing_date
-            last_ar_date = datetime.fromisoformat(last_ar_date)
-            if check_agm:
-                todo_start_date = (datetime(last_ar_date.year+1, 1, 1)).date()
-            else:
-                todo_start_date = (last_ar_date+datedelta.YEAR).date()
+            # Consider each filing as each year and add to find next ar year
+            next_ar_year += len(annual_report_filings)
```
Comment (kialj876) [inline legal-api/src/legal_api/resources/business/business_tasks.py:146]: if a business was founded in 2018 and they had filed their 2019+2020 annual reports (so business.last_ar_year would equal 2020), then their next_ar_year = 2020 + 2 = 2022. Am I missing something or does this logic need to change?

File: legal-api/src/legal_api/services/filings/validations/annual_report.py [hunk 2]
```diff
@@ -32,6 +31,28 @@ def requires_agm(business: Business) -> bool:
...
+def get_ar_dates(business: Business, start_date, next_ar_year):
+    """Get ar min and max date for the specific year."""
+    check_agm = requires_agm(business)
+    ar_min_date = datetime(next_ar_year, 1, 1).date()
+
+    # Make sure min date is greater than or equal to last_ar_date or founding_date
+    if ar_min_date < start_date:
+        ar_min_date = start_date
+
+    ar_max_date = datetime(next_ar_year, 12, 31).date()
```
Comment (kialj876) [inline legal-api/src/legal_api/services/filings/validations/annual_report.py:43]: this ar_max_date will need to be decided differently for extensions and is what I was talking about in the RFC. I think we need to have a design discussion with Thor and Lekshmi on what we'll add for ARs to make it work for extensions

**Review by kialj876 — COMMENTED** [review]
Hey Vysakh I think we should have a quick design session with thor and lekshmi before we move forward with this

File: legal-api/src/legal_api/resources/business/business_tasks.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/business/business_tasks.py:146]: `annual_report_filings` are the filings which are in not completed status, at that point `last_ar_year` will not have pending or draft filings (as per the functionality I dont see a possibility to have more than one in `annual_report_filings`). so in that case 2019 and 2020 will not be in `annual_report_filings`

Comment (kialj876) [reply legal-api/src/legal_api/resources/business/business_tasks.py:146]: ah okay you are right. That method name 'get_filings_by_type' is a bit misleading (it was probably me way back when lol) as it doesn't return completed filings

**Review by kialj876 — APPROVED** [review]

**Review by lekshmimallika-aot — APPROVED** [review]

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**